templates/repo/diff/box.tmpl

```diff
@@ -56,120 +64,135 @@
-							{{if $showFileViewToggle}}
-								<div class="ui compact icon buttons">
-									<span class="ui tiny basic button tooltip file-view-toggle" data-toggle-selector="#diff-source-{{$i}}" data-content="{{$.locale.Tr "repo.file_view_source"}}" data-position="bottom center">{{svg "octicon-code"}}</span>
-									<span class="ui tiny basic button tooltip file-view-toggle active" data-toggle-selector="#diff-rendered-{{$i}}" data-content="{{$.locale.Tr "repo.file_view_rendered"}}" data-position="bottom center">{{svg "octicon-file"}}</span>
-								</div>
-							{{end}}
-							{{if $file.IsProtected}}
-								<span class="ui basic label">{{$.locale.Tr "repo.diff.protected"}}</span>
-							{{end}}
-							{{if not (or $file.IsIncomplete $file.IsBin $file.IsSubmodule)}}
-								<a class="ui basic tiny button unescape-button">{{$.locale.Tr "repo.unescape_control_characters"}}</a>
-								<a class="ui basic tiny button escape-button" style="display: none;">{{$.locale.Tr "repo.escape_control_characters"}}</a>
-							{{end}}
-							{{if and (not $file.IsSubmodule) (not $.PageIsWiki)}}
-								{{if $file.IsDeleted}}
-									<a class="ui basic tiny button" rel="nofollow" href="{{$.BeforeSourcePath}}/{{PathEscapeSegments .Name}}">{{$.locale.Tr "repo.diff.view_file"}}</a>
-								{{else}}
-									<a class="ui basic tiny button" rel="nofollow" href="{{$.SourcePath}}/{{PathEscapeSegments .Name}}">{{$.locale.Tr "repo.diff.view_file"}}</a>
-								{{end}}
-							{{end}}
-							{{if and $.IsSigned $.PageIsPullFiles (not $.IsArchived)}}
-								{{if $file.HasChangedSinceLastReview}}
-									<span class="changed-since-last-review unselectable">{{$.locale.Tr "repo.pulls.has_changed_since_last_review"}}</span>
-								{{end}}
-								<div data-link="{{$.Issue.Link}}/viewed-files" data-headcommit="{{$.PullHeadCommitID}}" class="viewed-file-form unselectable{{if $file.IsViewed}} viewed-file-checked-form{{end}}">
-									<input type="checkbox" name="{{$file.GetDiffFileName}}" id="viewed-file-checkbox-{{$i}}" autocomplete="off" {{if $file.IsViewed}}checked{{end}}></input>
-									<label for="viewed-file-checkbox-{{$i}}">{{$.locale.Tr "repo.pulls.has_viewed_file"}}</label>
-								</div>
-							{{end}}
-						</div>
-					</h4>
-					<div class="diff-file-body ui attached unstackable table segment" {{if $file.IsViewed}}data-folded="true"{{end}}>
-						<div id="diff-source-{{$i}}" class="file-body file-code unicode-escaped code-diff{{if $.IsSplitStyle}} code-diff-split{{else}} code-diff-unified{{end}}{{if $showFileViewToggle}} hide{{end}}">
-							{{if or $file.IsIncomplete $file.IsBin}}
-								<div class="diff-file-body binary" style="padding: 5px 10px;">
-									{{if $file.IsIncomplete}}
-										{{if $file.IsIncompleteLineTooLong}}
-											{{$.locale.Tr "repo.diff.file_suppressed_line_too_long"}}
+		<div id="diff-container">
+				<script>
+					(() => {
+						window.config.pageData.pullRequestFileTree = {
```

Comment:
Maybe we could get them from DOM tree but not render second times in backend .

Reply:
Currently thinking about converting the file list into a Vue component and then only iterating the files once in backend.

Reply:
I've now converted the file list into a view component using the same data as file tree. So now we don't need a second iteration in our backend code.